types/rule_match.go

```diff
@@ -47,3 +51,9 @@ type MatchedRule interface {
 	AuditLog() string
 	ErrorLog() string
 }
+
+// RuleLogger is added to avoid breaking the Coraza v3.* API adding a Log() method to the MatchedRule interface
+// An assertion has to be done to check if the MatchedRule implements RuleLogger before calling Log()
+type RuleLogger interface {
```

Comment:
I'd rather not to expose this interface. Doing so we will be increasing the API surface and also this interface is just a workaround and very likely to be removed in v4. Also, go interface implementation is implicit so anyone can declare an interface and do the same assertion. any thoughts @anuraaga ?